src/dependency-version-checker.js

```diff
@@ -55,11 +58,28 @@ class DependencyVersionChecker {
     }
 
     if (!this.isAbove(compareVersion)) {
-      let error = new Error(message);
+      throw new SilentError(message);
+    }
+  }
 
-      error.suppressStacktrace = true;
+  isSingleton() {
+    if (this._isSingleton) {
+      return this._isSingleton;
+    }
+    let allInstances = discoverAddons(
+      getProject(this._parent._addon),
+      this.name
+    );
+    let uniqueInstances = new Set(allInstances.map(i => i.root));
+    return (this._isSingleton = uniqueInstances.size === 1);
+  }
 
-      throw error;
+  assertSingleton(_message) {
+    if (!this.isSingleton()) {
+      let message =
+        _message ||
+        `The addon \`${this._parent._addon.name}\` requires single version of ${this._type} package \`${this.name}\`, but there're multiple. Please resolve \`${this.name}\` to same version.`;
```

Comment:
I have one nit-pick:

In general, we like to prefix this errors with some concise context `[ember-cli]` or `[fastboot]` This helps users immediately know "what is complaining" In this context, it maybe it be `[${this._parent._addon.name}]`